num_enum_derive/src/lib.rs

```diff
@@ -288,9 +322,46 @@ impl Parse for EnumInfo {
                                                 die!(default.keyword =>
                                                     "Multiple variants marked `#[default]` or `#[num_enum(default)]` found"
                                                 );
+                                            } else if has_catch_all_variant {
+                                                die!(default.keyword =>
+                                                    "Attribute `default` is mutually exclusive with `catch_all`"
+                                                );
                                             }
                                             attr_spans.default.push(default.span());
                                             is_default = true;
+                                            has_default_variant = true;
+                                        }
+                                        NumEnumVariantAttributeItem::CatchAll(catch_all) => {
+                                            match variant
+                                                .fields
+                                                .iter()
+                                                .collect::<Vec<_>>()
+                                                .as_slice()
+                                            {
+                                                _ if has_catch_all_variant => {
```

Comment:
It's a little strange to have these guarded catch-all branches in a match - could we pull these out to before the match?

Reply:
I did it like this to have all errors in one place, but I agree that it is a bit strange. I changed it back to an if else if.